server/src/service.rs

```diff
@@ -134,35 +139,52 @@ impl Service {
             manager: Default::default(),
             connection: connection.clone(),
         };
-        let collections = service.fetch_collections().await;
 
+        // Load login/default collection
+        let mut login: Option<Collection> = None;
+        if secret.is_some() {
+            login = Some(service.fetch_login(secret).await?);
+        }
+        // Load session collection
+        let session = service.fetch_session().await?;
+
+        // Dispatch Service
         object_server
             .at(oo7::dbus::api::Service::PATH.as_deref().unwrap(), service)
             .await?;
-
-        for collection in collections {
-            object_server
-                .at(collection.path().clone(), collection)
-                .await?;
+        // Dispatch login collection
+        if login.is_some() {
+            let login = login.unwrap();
+            object_server.at(login.path().clone(), login).await?;
         }
+        // Dispatch session collection
+        object_server.at(session.path().clone(), session).await?;
 
         Ok(())
     }
 
-    async fn fetch_collections(&self) -> Vec<Collection> {
-        let mut collections = Vec::new();
-        // todo: create default collection
-
-        // create temporary session collection
-        let session_collection = Collection::new("session", "session", Arc::clone(&self.manager));
-        collections.push(session_collection);
+    pub async fn fetch_session(&self) -> Result<Collection, Error> {
+        let secret = Secret::random();
+        let session = Collection::new(
+            "session",
+            "session",
+            Arc::clone(&self.manager),
+            Arc::new(Keyring::temporary(secret).await?),
+        );
+        self.collections.lock().await.push(session.path().clone());
+
+        Ok(session)
+    }
 
-        let mut lock = self.collections.lock().await;
-        for collection in &collections {
-            lock.push(collection.path().clone());
-        }
-        drop(lock);
+    pub async fn fetch_login(&self, secret: Option<Secret>) -> Result<Collection, Error> {
```

Comment:
It is not called login anywhere except it alias. So just name it fetch_default_collection or so

Reply:
Also, why are you passing Option<Secret> here, it seems useless... just to unwrap it later...